webpack.common.js

```diff
@@ -0,0 +1,41 @@
+const path = require('path');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
+
+const isDev = process.env.NODE_ENV === 'development';
+
+module.exports = {
+  mode: process.env.NODE_ENV,
+  entry: './src/index.js',
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: isDev ? 'bundle.js' : 'bundle.min.js',
+  },
+  plugins: [
+    new HtmlWebpackPlugin({ template: './src/index.html' }),
+    new CleanWebpackPlugin(),
+    new MiniCssExtractPlugin({ filename: isDev ? 'styles.css' : 'styles.min.css' }),
+  ],
+  resolve: { extensions: ['.js', '.jsx'] },
```

Comment:
have a look on https://webpack.js.org/configuration/resolve/#resolvemodules
What are your thoughts about `modules` prop? Do you think we can benefit from it?